src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```
@@ -139,7 +140,77 @@ public void RollBackWorkloadPackInstall(PackInfo packInfo, SdkFeatureBand sdkFea
+                   {
+                       packagePath = _nugetPackageDownloader.DownloadPackageAsync(WorkloadManifestUpdater.GetManifestPackageId(sdkFeatureBand, manifestId), new NuGetVersion(manifestVersion.ToString())).Result;
+                       tempExtractionDir = Path.Combine(_tempPackagesDir.Value, $"{manifestId}-{manifestVersion}-extracted");
+                       Directory.CreateDirectory(tempExtractionDir);
+                       var manifestFiles = _nugetPackageDownloader.ExtractPackageAsync(packagePath, tempExtractionDir).Result;
+
+                       if (Directory.Exists(manifestPath) && Directory.GetFileSystemEntries(manifestPath).Any())
+                       {
+                           // Backup existing manifest data for roll back purposes
+                           tempBackupDir = Path.Combine(_tempPackagesDir.Value, $"{manifestId}-{manifestVersion}-backup");
+                           FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(manifestPath, tempBackupDir));
+                           Directory.Delete(manifestPath, true);
+                       }
+
+                       Directory.CreateDirectory(Path.GetDirectoryName(manifestPath));
+                       FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(tempExtractionDir, manifestPath));
+                   },
+                    rollback: () => {
+                        if (!string.IsNullOrEmpty(tempBackupDir))
+                        {
+                            FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(tempBackupDir, manifestPath));
+                        }
+                    });
+            }
+            catch (Exception e)
+            {
+                throw new Exception(string.Format(LocalizableStrings.FailedToInstallWorkloadManifest, manifestId, manifestVersion, e.Message));
+            }
+            finally
+            {
+                // Delete leftover dirs and files
```

nit: this will cause a concurrent issue if there are 2 dotnet process running at the sametime. I prefer try out best to delete the empty folder. If not, the system should be able to tolerate empty folder.

---

What do you mean by try our best to delete the empty folder? Move this to the try clause?

---

the delete could fail, catch the exception and swallow it

---

put delete in to try part instead of finally